### ISSUE TITLE
Node/Http should support https

### ISSUE THREAD

**Issue by mtornwall** [Starcounter/Starcounter#2954]
As it stands right now, an attempt to use a URL with the `https://` prefix with `Http.GET` (or the `Node` functions) produces an `ArgumentOutOfRange` exception. For my purposes, https support is non-optional, and I believe I am not alone in this.

I realize that implementing support for SSL/TLS may require a significant amount of work. Perhaps there should be an officially sanctioned solution in the meantime? :-)

If anyone else is having this issue I have a workaround that involves running a `System.Net.Http.HttpClient` request in the background using `Task.Run` and then scheduling the callback on the calling Starcounter thread using `DbSession.RunAsync`.

**Comment by dan31** [comment]
Hi @mtornwall! @alemoi, WDYT?

**Comment by warpech** [comment]
@mtornwall if I understand correctly, the actual TLS part is done by nginx in front of Starcounter. Just thinking, isn't it possible for nginx to rewrite the URL so that Starcounter is called with `http://`?

If not then we might need to allow `https://` handlers before we have TLS support.

**Comment by mtornwall** [comment]
@warpech Yes, when Starcounter is taking on the role of server this is how it works. But my Starcounter apps need to make external HTTP calls, and to do this they use `Http.GET` et al.

**Comment by warpech** [comment]
I didn't notice this issue is for outbound calls. This makes my comment irrelevant, sorry!
